core/src/main/java/hudson/model/LoadStatistics.java

```diff
@@ -67,15 +69,59 @@
 @ExportedBean
 public abstract class LoadStatistics {
     /**
+     * {@code true} if and only if the new way of building statistics has been implemented by this class.
+     * @since FIXME
+     */
+    private final boolean modern;
+
+    /**
+     * Number of executors defined for Jenkins and how it changes over time.
+     * @since FIXME
+     */
+    @Exported
+    public final MultiStageTimeSeries definedExecutors;
+
+    /**
+     * Number of executors on-line and how it changes over time. Replaces {@link #totalExecutors}
+     * @since FIXME
+     */
+    @Exported
+    public final MultiStageTimeSeries onlineExecutors;
+
+    /**
+     * Number of executors in the process of coming on-line and how it changes over time.
+     * @since FIXME
+     */
+    @Exported
+    public final MultiStageTimeSeries connectingExecutors;
+
+    /**
      * Number of busy executors and how it changes over time.
      */
     @Exported
     public final MultiStageTimeSeries busyExecutors;
 
     /**
+     * Number of executors not executing and how it changes over time. Note the these executors may not be able to
+     * take work, see {@link #availableExecutors}.
+     * @since FIXME
+     */
+    @Exported
+    public final MultiStageTimeSeries idleExecutors;
+
+    /**
+     * Number of executors not executing and available to take work and how it changes over time.
+     * @since FIXME
+     */
+    @Exported
+    public final MultiStageTimeSeries availableExecutors;
+
+    /**
      * Number of total executors and how it changes over time.
+     * @deprecated use {@link #onlineExecutors}
```

Comment:
Doc should point out `totalExecutors==onlineExecutors` now for backward compatibility support.